packages/suite-desktop/src/support/screens/TorLoadingScreen.tsx

```diff
@@ -0,0 +1,238 @@
+    width: 100%;
+    min-height: 45px;
+`;
+
+const ProgressMessage = styled.div`
+    margin-left: 10px;
+    margin-right: 10px;
+    color: ${props => props.theme.TYPE_DARK_GREY};
+`;
+
+const InfoWrapper = styled.div`
+    display: flex;
+    justify-content: space-between;
+`;
+
+const MessageWrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    padding-bottom: 50px;
+`;
+
+const Separator = styled.hr`
+    height: 1px;
+    width: 100%;
+    background: none;
+    border: 0;
+    border-top: 1px solid ${props => props.theme.STROKE_GREY};
+    margin: 20px;
+`;
+
+const TryAgainWrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+`;
+
+const DisableButton = styled(Button)`
+    padding-right: 30px;
+    padding-left: 30px;
+    border: 1px solid #f4f4f4;
+    margin-left: 10px;
+`;
+
+interface TorLoadingScreenProps {
+    callback: (value?: unknown) => void;
+}
+
+export const TorLoadingScreen = ({ callback }: TorLoadingScreenProps) => {
+    const [torStatus, setTorStatus] = useState<TorStatus>(TorStatus.Enabling);
+    const [progress, setProgress] = useState<number>(0);
+
+    useEffect(() => {
+        desktopApi.on('tor/bootstrap', (bootstrapEvent: BootstrapTorEvent) => {
+            if (torStatus === TorStatus.Disabling) return;
+            if (bootstrapEvent.type === 'error') {
+                setTorStatus(TorStatus.Error);
+            }
+            if (bootstrapEvent.type !== 'progress') return;
+            if (
+                bootstrapEvent &&
+                bootstrapEvent.type === 'progress' &&
+                bootstrapEvent.progress &&
+                bootstrapEvent.progress.current
+            ) {
+                setProgress(bootstrapEvent.progress.current);
+                if (bootstrapEvent.progress.current === bootstrapEvent.progress.total) {
+                    setTorStatus(TorStatus.Enabled);
+                    callback();
+                } else {
+                    setTorStatus(TorStatus.Enabling);
+                }
+            }
+        });
+        return () => desktopApi.removeAllListeners('tor/bootstrap');
+    }, [torStatus, callback]);
+
+    let message = 'Enabling Tor';
+    if (torStatus === TorStatus.Error) {
+        message = 'Enabling Tor Failed';
+    } else if (torStatus === TorStatus.Disabling) {
+        message = 'Disabling Tor';
+    }
+
+    const tryAgain = async () => {
+        setProgress(0);
+        setTorStatus(TorStatus.Enabling);
+        const torLoaded = await desktopApi.toggleTor(true);
+        if (!torLoaded.success) {
+            setTorStatus(TorStatus.Error);
+        }
+    };
+
+    const disableTor = async () => {
+        let fakeProgress = 0;
+        setTorStatus(TorStatus.Disabling);
+        desktopApi.removeAllListeners('tor/bootstrap');
+        desktopApi.toggleTor(false);
+
+        await new Promise(callback => {
```

Comment:
Do I understand it correctly that this is totally fake loading for turning Tor off, because otherwise it would be too quick for the user to notice?

Also nit: I'd better rename the `callback` inside the Promise to `resolve`, so it differs from the other `callback` 
Also tiny nit (oh my god, it's a palindrome!): `fakeProgress` could be hidden inside Promise callback

Reply:
Yes! Name changed!